src/utils/bufferCodec.ts

```diff
@@ -0,0 +1,15 @@
+import { TextDecoder, TextEncoder } from 'web-encoding'
+
+export function encodeBuf(input: string): ArrayBuffer {
+  const encoder = new TextEncoder()
+  const encoded = encoder.encode(input)
+  return encoded.buffer.slice(
+    encoded.byteOffset,
+    encoded.byteOffset + encoded.byteLength
+  )
+}
+
+export function decodeBuf(input: ArrayBuffer): string {
+  const decoder = new TextDecoder('utf-8')
```

Comment:
I think we should accept the buffer encoding as an input to this function. We should respect the original request's encoding. What do you think about this?

Reply:
We can provide that and also accept it in `text` function

Reply:
The `text` should return the text representation of the given buffer with the encoding that was initially given to that request instance (we can fallback to utf8 if not set). If you wish to create a text representation with a different encoding you should do `req.arrayBuffer` and then something like `.toString('latin1')` based on the ArrayBuffer API (you get the idea). 

I'm trying to say that accepting a custom encoding in the `text` function is not idiomatic.